packages/functional/tests/ui/desktop/e2e/shipping.spec.ts

```diff
@@ -1,47 +1,56 @@
-import { expect, test } from '@playwright/test';
+import { expect, Page, test } from '@playwright/test';
 
-import { CartActions } from '../../../../actions/cart-actions';
-import { ProductActions } from '../../../../actions/product-actions';
+async function addEstimatedShippingCosts(
+  page: Page,
```

Comment:
There's really only two ways for me to do that as the `page` object is only within the `test` arguments closure.

1. By extending the test and adding a fixture to be used:
```ts
const test = base.extend<{ addEstimatedShippingCosts: AddEstimatedShippingCosts }>({
  addEstimatedShippingCosts: async ({ page }, use) => {
    await use(async (country: string, state: string, city: string, zip: string) => {
      // ...
    });
  },
});

test('...', async ({ page, addEstimatedShippingCosts }) => {...}
```

2. By binding the `page` object to the functions `this`
```ts
async function addEstimatedShippingCosts(
  this: Page,
  country: string,
  state: string,
  city: string,
  zip: string,
): Promise<void> {}

await addEstimatedShippingCosts.call(page, 'United States', 'Texas', 'Austin', '76267');
```

I think either of these might be a bit too much, especially 2. so I'll probably leave as is. The only optimization I would probably make here is by having the arguments be inside an object instead of individual args but I will save that for another time e.g.
```ts
interface Options {
  country: string;
  state: string;
  city: string;
  zip: string;
}

async function addEstimatedShippingCosts(page: Page, options: Options): Promise<void> {
  const { country, state, city, zip } = options;
  // ...
}
```

Reply:
Ah, I thought it was simpler. I thought we were within the test context and didn't realize function was out of context for test although it was within the same file 🤦